server/controllers/UserController.js

```diff
@@ -155,10 +142,45 @@ export default class UserController {
       await User.update({
         ...body
       }, { where: { id } });
-      return responseMessage(response, 200, { status: 'success', message: 'update successful' });
+      return responseMessage(response, 200, { status: 'success', message: 'profile update successful' });
     } catch (error) {
       /* istanbul ignore next */
       responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+   * @description allows users to borrow a book
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof UserController
+   */
+  static async borrowBook(request, response) {
+    try {
+      const { id, paymentStatus } = request.userData;
+      const { bookId } = request.body;
+      if (!paymentStatus) responseMessage(response, 402, { message: 'insufficient funds to borrow book' });
+      const existingBook = await findBook(bookId, response);
+      const { dataValues, BookAuthors: [BookAuthor] } = existingBook;
+      const { authorName } = BookAuthor.dataValues.Author;
+      delete dataValues.BookAuthors;
+      const [borrowedBook, bookDuration] = await findLendingHistory(id, bookId, response);
+      if (borrowedBook && bookDuration === '') responseMessage(response, 409, { message: 'book already borrowed' });
```

Comment:
You need to use the 'return' keyword on this line otherwise it will lead to this error: _Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client_

Reply:
Noted